client/src/lib/utils/insertMessage.ts

```
@@ -0,0 +1,63 @@
+
+import { createPublicThread }  from "./createPublicThread";
+import { MessageCardProps } from "../../ui/messaging/MessageCard";
+import { Profile } from "../store/profile";
+import { createSupabaseClient } from "../supabase/client"
+
+
+export async function insertMessage(newMessage: MessageCardProps, conversationId: string, profile: Profile) 
+{
+    const supabase = createSupabaseClient()  
+
+    if (conversationId && newMessage) {
+      const { data: publicThreadData, error: publicThreadError } = await supabase
+        .from("public_user_conversation")
+        .select("thread_id")
+        .eq("conversation_id", conversationId)
+        .eq("sender_id", profile.profile_id);
+
+      // Log any errors encountered while checking for public threads
+      if (publicThreadError) {
```

can you return the error so I can render the error message in the UI